### PR TITLE
[quant] Add op support for linear_relu_dynamic_fp16

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63826
* __->__ #63824
* #63820
* #63799

Summary:
Add a fused operator implementation that will work with the quantization fusion APIs.
Once FBGEMM FP16 kernel supports relu fusion natively we can remove the addition from the PT operator.

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30503514](https://our.internmc.facebook.com/intern/diff/D30503514)

### CODE REVIEW

**Comment by supriyar** [comment]
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30503514).

File: test/quantization/core/test_quantized_op.py [hunk 1]
```diff
@@ -2778,6 +2778,36 @@ def test_qlinear_legacy(self, batch_size, input_channels, output_channels):
         self.assertEqual(Y_fp32, Y_fp32_ref,
                          msg="torch.ops.quantized.fbgemm_linear_dynamic results are off")
 
+    @skipIfNoFBGEMM
+    @given(
```
Comment (jerryzh168) [inline test/quantization/core/test_quantized_op.py:2782]: should we use loops instead of hypoothsis here?

**Review by jerryzh168 — APPROVED** [review]
looks good, I think we should probably remove hypothesis

**Comment by supriyar** [comment]
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30503514).

File: test/quantization/core/test_quantized_op.py (hunk 1 above)
Comment (supriyar) [reply test/quantization/core/test_quantized_op.py]: removed it :)

**Pull request closed without merging**